File: app/src/main/java/com/njust/helper/settings/AboutActivity.java
package com.njust.helper.settings;

import android.content.ActivityNotFoundException;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.support.v4.view.MenuItemCompat;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.ShareActionProvider;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.njust.helper.R;
import com.njust.helper.activity.BaseActivity;

/**
 * 关于
 *
 * @author zwb
 */
public class AboutActivity extends BaseActivity {
    @Override
    protected int layoutRes() {
        return R.layout.activity_about;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.about, menu);
        MenuItem item = menu.findItem(R.id.item_share);
        ShareActionProvider provider = new ShareActionProvider(this);
        Intent shareIntent = new Intent(Intent.ACTION_SEND)
                .setType("text/plain")
                .putExtra(Intent.EXTRA_TEXT, getString(R.string.share_about));
        provider.setShareIntent(shareIntent);
        MenuItemCompat.setActionProvider(item, provider);
        return true;
    }

    public void feedback(View view) {
        new AlertDialog.Builder(this)
                .setTitle("意见反馈")
                .setMessage("您可以加入QQ群217887769进行反馈")
                .setPositiveButton("点击直接加群", (dialog, which) -> {
                    Intent intent = new Intent(Intent.ACTION_VIEW,
                            Uri.parse("http://jq.qq.com/?_wv=1027&k=2HCZ1MK"));
                    startActivity(intent);
                })
                .setNegativeButton("返回", null)
                .show();
    }

    public void comment(View view) {
        try {
            Uri uri = Uri.parse("market://details?id=" + getPackageName());
            Intent intent = new Intent(Intent.ACTION_VIEW, uri);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            startActivity(intent);
        } catch (ActivityNotFoundException e) {
            showSnack("未找到应用市场");
        }
    }

    public void updateLog(View view) {
        UpdateLogDialog.showUpdateDialog(this);
    }
}

File: app/src/main/java/com/njust/helper/settings/AboutActivity.kt
package com.njust.helper.settings

import android.content.ActivityNotFoundException
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.njust.helper.R
import com.njust.helper.update.UpdateLogDialog
import com.zwb.commonlibs.utils.requireSystemService
import kotlinx.coroutines.flow.MutableSharedFlow

/**
 * 关于
 *
 * @author zwb
 */
class AboutActivity : AppCompatActivity() {
  private val snackbarMessageFlow = MutableSharedFlow<String>()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContent {
      AboutScreen(
        onClickHome = { finish() },
        onClickFeedback = { onClickFeedback() },
        onClickComment = { onClickComment() },
        onClickUpdateLog = { onClickUpdateLog() },
        onClickShare = { onClickShare() },
        snackbarMessageFlow = snackbarMessageFlow,
      )
    }
  }

  private fun onClickFeedback() {
    AlertDialog.Builder(this)
      .setTitle("意见反馈")
      .setMessage("您可以加入QQ群${QQ_GROUP_ID}进行反馈")
      .setPositiveButton("点击直接加群") { _, _ ->
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(QQ_GROUP_URI))
        startActivity(intent)
      }
      .setNeutralButton("复制群号") { _, _ ->
        val clipboardManager = requireSystemService<ClipboardManager>()
        clipboardManager.setPrimaryClip(ClipData.newPlainText(null, QQ_GROUP_ID))
        }
        .setNegativeButton("返回", null)
        .show()
  }

  private fun onClickComment() {
    try {
      val uri = Uri.parse("market://details?id=$packageName")
      val intent = Intent(Intent.ACTION_VIEW, uri)
      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
      startActivity(intent)
    } catch (e: ActivityNotFoundException) {
      lifecycleScope.launchWhenResumed {
        snackbarMessageFlow.emit("未找到应用市场")
      }
    }
  }

  private fun onClickUpdateLog() {
    UpdateLogDialog.showUpdateDialog(this)
  }

  private fun onClickShare() {
    val shareIntent = Intent(Intent.ACTION_SEND)
      .setType("text/plain")
      .putExtra(Intent.EXTRA_TEXT, getString(R.string.share_about))
    try {
      startActivity(shareIntent)
    } catch (e: ActivityNotFoundException) {
      lifecycleScope.launchWhenResumed {
        snackbarMessageFlow.emit("未找到可分享的应用")
      }
    }
  }
}

private const val QQ_GROUP_ID = "217887769"
private const val QQ_GROUP_URI = "http://jq.qq.com/?_wv=1027&k=2HCZ1MK"
